doc/source/examples/metrology/Creating Network Definitions for SOLT Calibration Standards.ipynb

```diff
@@ -0,0 +1,744 @@
+    "\n",
+    "Let's translate these formulas to code."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def offset_gamma_and_zc(offset_delay, offset_loss, offset_z0=50):\n",
+    "    alpha_l = (offset_loss * offset_delay) / (2 * offset_z0)\n",
+    "    alpha_l *= np.sqrt(freq.f / 1e9)\n",
+    "    beta_l = 2 * np.pi * freq.f * offset_delay + alpha_l\n",
+    "    gamma_l = alpha_l + 1j * beta_l\n",
+    "    zc = (offset_z0) + (1 - 1j) * (offset_loss / (2 * 2 * np.pi * freq.f)) * np.sqrt(freq.f / 1e9)\n",
+    "    return gamma_l, zc"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note how the broadcasting feature in `numpy` is used here. The quantities\n",
+    "`alpha_l`, `beta_l`, and `zc` are all frequency-dependent, thus they're\n",
+    "arrays, not scalars. But instead of looping over each frequency explicitly\n",
+    "and adding them to an array, here, arrays are automatically created by the\n",
+    "multiplication of a scalar and a `numpy.array`. We'll continue to use this\n",
+    "technique in the following code.\n",
+    "\n",
+    "Next, an offset transmission line can be defined using `DefinedGammaZ0`. Note the\n",
+    "`Z` is `Z0` must be a capital letter. Writing `z0` would define a port impedance\n",
+    "instead of a line impedance. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gamma_l, zc = offset_gamma_and_zc(29.242e-12, 2.2e9)\n",
+    "medium_open = DefinedGammaZ0(frequency=freq, gamma=gamma_l, Z0=zc)\n",
+    "\n",
+    "gamma_l, zc = offset_gamma_and_zc(31.785e-12, 2.36e9)\n",
+    "medium_short = DefinedGammaZ0(frequency=freq, gamma=gamma_l, Z0=zc)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "What we have defined is just a physical medium, not the transmission\n",
+    "line itself. To complete our definition, we must define a line with a\n",
+    "specific length. All we need to do here is to define a line with *unity*\n",
+    "length (because $\\gamma l$ is used as $\\gamma$). Since the unit of\n",
+    "measurement is meter, we simply define an \"one meter\" line.\n",
+    "\n",
+    "It should be noted that the class `DefinedGammaZ0` is supposed to\n",
+    "represent an actual medium with some physical properties, and the\n",
+    "`line()` method is supposed to define a line with a real physical\n",
+    "length. But due to the peculiar way we're using it, it's solely\n",
+    "a calculation tool to obtain the S-parameters of the offset\n",
+    "transimission line. The medium itself is physically meaningless."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "line_open = medium_open.line(d=1, unit='m')\n",
```

Comment:
This creates a line with complex characteristic and port impedance. The complex port impedance can be a problem if the S-parameters are passed to the calibration routine which usually assume a fixed real impedance. Also if the S-parameters are exported as touchstone, the S-parameters are not renormalized to a fixed real impedance and the exported network is incorrect. Unfortunately creating a line with characteristic impedance of the medium and given port impedance is not very intuitive (Issue #651). You should set the port impedance `z0` in the media and then generate the line with `z0=medium_open.Z0, embed=True`. It probably doesn't cause that large error in practice since the imaginary part of the characteristic impedance should be quite close to zero.

If you check the `z0` property of the generated standards its complex because of the input port is from this transmission line.

The same issue applies to lines in the later example.

Reply:
> It probably doesn't cause that large error in practice since the imaginary part of the characteristic impedance should be quite close to zero.

Thanks for the comment. I'm doing some final tests, one of them is generating a bunch of random parameters and compare SignalIntegrity and my results. I noticed a small numerical disagreement up to 0.006 in the offset line model (I thought it has been solved but apparently not really). This is probably the reason.

Reply:
Yes! Setting `embed=True` and now I see a 10x reduction in numerical disagreement.

Reply:
Solved.